src/domain/synthetics/common/useUserIncentiveData.ts

```diff
@@ -0,0 +1,59 @@
+import { gql } from "@apollo/client";
+import { getSyntheticsGraphClient } from "lib/subgraph";
+import useSWR from "swr";
+
+type UserIncentiveData = {
+  id: string;
+  typeId: string;
+  amounts: string[];
+  amountsInUsd: string[];
+  blockNumber: string;
+  receiver: string;
+  timestamp: number;
+  tokens: string[];
+  transactionHash: string;
+};
+
+const USER_INCENTIVE_QUERY = gql`
+  query userIncentiveData($account: String!) {
+    distributions(
+      orderBy: timestamp
+      orderDirection: desc
+      where: { receiver: $account, typeId_in: [1001, 1002, 1003] }
+      first: 1000
+    ) {
+      typeId
+      amounts
+      amountsInUsd
+      blockNumber
+      id
+      receiver
+      timestamp
+      tokens
+      transactionHash
+    }
+  }
+`;
+
+export default function useUserIncentiveData(chainId: number, account?: string) {
+  const graphClient = getSyntheticsGraphClient(chainId);
+  const userIncentiveDataCacheKey =
+    chainId && graphClient && account ? [chainId, "useUserIncentiveData", account] : null;
+
+  async function fetchUserIncentiveData(): Promise<UserIncentiveData[]> {
+    if (!account) {
+      return [];
+    }
+
+    const response = await graphClient!.query({
+      query: USER_INCENTIVE_QUERY,
+      variables: { account: account.toLowerCase() },
+    });
```

Comment:
would be better to use `getNormalizedIncentive` right here, once request is done